20/faq/feat

ccoutts2 opened the pull request:
- Utilised dropdown component in FAQ section
- Refactored dropdown section to accept different data from different components
- Used placeholder text from current site - i suspect the content will change for the new site
- Added FAQ link in about page

ccoutts2 committed:
WIP - add FAQ page and link in footer

ccoutts2 committed:
Merge branch 'develop' into 20/FAQ/feat

ccoutts2 committed:
Refactor dropdown component to be more reusable

ccoutts2 committed:
Add FAQ link to about page

ccoutts2 committed:
Fix typo on about page

ccoutts2 committed:
Fix typo

kabirseth4 commented on src/app/faq/data.ts:
```diff
@@ -0,0 +1,43 @@
+interface detailsProps {
```
Import from component

kabirseth4 commented on src/app/faq/page.tsx:
```diff
@@ -0,0 +1,31 @@
+const page = () => {
+  return (
+    <section className="w-full">
+      <PageHeader header="FAQ's" />
```
No apostrophe

kabirseth4 commented on src/components/DetailDropdown/DetailDropdown.tsx:
```diff
@@ -5,32 +5,19 @@ import { useGSAP } from "@gsap/react";
-      "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Asperiores dolores aut hic nulla voluptatem harum inventore. Dolorum, voluptatibus hic, ad fuga quae quisquam sit aliquam porro dignissimos eius nulla consectetur.",
-  },
-];
+interface DetailDropdown {
```
Should have different name from component, DetailDropdownProps

kabirseth4 commented on src/components/DetailDropdown/DetailDropdown.tsx:
```diff
@@ -5,32 +5,19 @@ import { useGSAP } from "@gsap/react";
 import { PlusMinusButton } from "../buttons/PlusMinusButton/PlusMinusButton";
 
-interface productDetailsProps {
+interface DetailsProps {
```
Should export and import this interface to faq/data.ts instead of remaking. Also don't think it should have Props in the name, since it's not props

kabirseth4 commented on src/components/Footer/Footer.tsx:
```diff
@@ -45,20 +46,28 @@ const Footer = () => {
-          <p className="text-xs lg:text-base">kitchen</p>
-          <p className="text-xs lg:text-base">living room</p>
+          <ul className="text-center md:text-right">
+            <li className="text-xs lg:text-base">bedroom</li>
+            <li className="text-xs lg:text-base">bathroom</li>
+            <li className="text-xs lg:text-base">kitchen</li>
+            <li className="text-xs lg:text-base">living room</li>
```
Assume these are all placeholders, if so I don't think we need them as there are only bedroom schemes to start anyway

ccoutts2 replied:
Yup - placeholder. Will make more sense to map through the list as well but figured once we refactor site as ultimately things will change

kabirseth4 commented on src/components/Footer/Footer.tsx:
```diff
@@ -1,6 +1,7 @@
 "use client";
 import React, { useState } from "react";
 import styles from "./Footer.module.scss";
```
Unused import?

kabirseth4 commented on src/components/sections/SchemeDetails/data.ts:
```diff
@@ -0,0 +1,22 @@
+interface detailsProps {
```
Import from component

kabirseth4 reviewed (changes requested):
Couple changes :)

ccoutts2 committed:
Make changes as per PR feedback

ccoutts2 commented:
@kabirseth4 

Changes

- Removed styles.module import in Footer
- Changed interface name to DetailItem
- Export and imported DetailItem into relevant files

kabirseth4 commented on src/app/faq/data.ts:
```diff
@@ -1,9 +1,9 @@
-interface detailsProps {
+export interface DetailItem {
```
Doesn't really make sense to declare and export it from this file. I think create a types folder in src and export it from a file called index.d.ts

ccoutts2 replied:
I'd be inclined to call it a types.ts file within the FAQ folder, where they're exported from. Let me know if you'd rather your suggestion @kabirseth4 

Update, I have done my way but happy to change

ccoutts2 committed:
Merge branch 'develop' into 20/FAQ/feat

ccoutts2 committed:
Move DetailItem type to types.ts folder

ccoutts2 committed:
Create Accordion component and atomic components, remove detailsdropdown component

ccoutts2 commented:
@kabirseth4 

- Removed the DetailsDropdown component which used GSAP to animate the dropdown
- Added Accordion component and atomic components which is more accessible than using GSAP
- Changed speed of PlusMinusButton animation
